airbyte-integrations/connector-templates/generator/generate.sh

```diff
@@ -3,7 +3,7 @@
 error_handler() {
   echo "While trying to generate a connector, an error occurred on line $1 of generate.sh and the process aborted early.  This is probably a bug."
 }
-trap 'error_handler $LINENO' ERR
+trap 'error_handler $LINENO' EXIT
```

Comment:
This doesn't do what you'd hope.  ERR traps when the prior line was exit code nonzero.  EXIT traps any time the script exits, including when it exits successfully with a zero exit code.  That means the script can run completely fine and it'll still print the error message after, confusing the user.  This has to stay ERR, or be refactored to use a different handling strategy entirely.

Reply:
@jennybrown8 is the suggestion made in this [SO post](https://stackoverflow.com/a/58297505/4195169) a reasonable approach?

If my understanding of the problem is correct, the issue is that wherever this generate.sh script is getting run the `ERR` signal does not exist but `EXIT` does. So trapping exit but only emitting a non zero code upon checking the exit state solves the problem? Maybe there is a better way of doing this?

Very funny that this is a script that runs a docker container. We put the generation code in the docker container to avoid these portability issues... then sure enough the script has a portability issue. (@sherifnada you'll think this is funny)

Reply:
This PR doesn't say what problem it's trying to solve.  What was the problem with ERR in the first place?

Reply:
> "Try to run connector template got trap: ERR: bad trap. Users are complaining about this."

I ran `generate.sh` on Ubuntu 20.04 and got the same problem as reported by user because of this I tried to suggest this modification. After the change I ran the script again and the ERR messages was gone. The script run correctly both ways, but with `trap ERR` the message is prompted to user.
![Screenshot from 2021-08-24 20-56-54](https://user-images.githubusercontent.com/5154322/130704869-07f6b52a-6dc9-43a6-a819-e6f3a3de6b13.png)